CMakeLists.txt

```diff
@@ -22,6 +22,9 @@ if(NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   message(FATAL_ERROR "In-source builds are not allowed. Please make a new directory (called a build directory) and run CMake from there.")
 endif()
 
+# If there are multiple python env we need to explicitly set the python3 executable
+set(Python3_EXECUTABLE $ENV{TTMLIR_VENV_DIR}/bin/python3)
```

Comment:
Does this actually do what we think it does? `Python3_EXECUTABLE` is a result variable that cmake populates https://cmake.org/cmake/help/latest/module/FindPython3.html#result-variables.

It might just get clobbered by cmake when MLIR runs `FindPython3`.

Reply:
This actually fixes FindPython3. If I remove this line it will return python from pybuda_toolchain not tt_mlir_toolchain. It's actually recommended to override it by MLIR [docs](https://mlir.llvm.org/docs/Bindings/Python/#cmake-variables).